testshared/src/main/java/org/odk/collect/testshared/FakeScheduler.kt

```diff
@@ -3,58 +3,72 @@ package org.odk.collect.testshared
 import org.odk.collect.async.Cancellable
 import org.odk.collect.async.Scheduler
 import org.odk.collect.async.TaskSpec
+import java.util.LinkedList
 import java.util.function.Consumer
 import java.util.function.Supplier
 
 class FakeScheduler : Scheduler {
```

Comment:
For some of the tests this needed to be a bit more advanced so it now actually keeps scheduled tasks in queues/lists so it runs them in the correct order and can run repeated tasks.